src/emc/usr_intf/sockets.c

```diff
@@ -306,7 +306,7 @@ int sockPrintfError(int fd, const char *format, .../*args*/ )
   va_list ap;
   int size = 0;
 
-  strncpy(buf, huh, sizeof(huh)); // note: sizeof(huh) < MAXMSG
+  strncpy(buf, huh, sizeof(buf));
```

Comment:
I looked at this for a while as well, and I think the old code is correct, but its a not obvious what it does.
Its supposed to copy huh to buf, so len of huh is correct.

Reply:
strncpy copies at most size bytes, and stops at the /0. So The effect should be exactly the same, except in cases where (for some reason) huh became longer than buf.

Reply:
in strncpy/strlcpy the size should always be "about" the destination buffer, even if the source & destination are demonstrably the same size or the source is demonstrably shorter.